libfreerdp/core/gateway/rdg.c

```diff
@@ -1583,6 +1583,17 @@ static long rdg_bio_ctrl(BIO* bio, int cmd, long arg1, void* arg2)
 		else
 			status = 1;
 	}
+	else if (cmd == BIO_C_GET_FD)
+	{
+		/*
+		 * Even if two FDs are part of RDG, only one FD can be returned here.
+		 *
+		 * In FreeRDP, BIO FDs are only used for polling, so it is safe to use the outgoing FD only
+		 *
+		 * See issue #3602
+		 */
+		status = BIO_ctrl(tlsOut->bio, cmd, arg1, arg2);
+	}
 
 	return status;
```

Comment:
Makes sense to initialize "status" with "-1" on line 1529 to fail for non implemented commands.

P.S. I can't leave comment for line 1529 for some reason.

Reply:
I made a test run and it seems now all commands required for the test run are handled in this function. The only difference would be in the command `BIO_C_GET_EVENT`, where `status` can be left untouched (which did not happen in my test run).

I agree, Initializing with -1 would be a reasonable future-proof improvement. @akallabeth any objections? Or is the PR good to go? :)

Reply:
Looks like, handling of BIO_C_GET_EVENT also needs to be changed to
    status = BIO_ctrl(tlsOut->bio, cmd, arg1, arg2);

Reply:
@p-pautov Yes, you're right, that will be required to get this stuff working on windows

Reply:
@p-pautov @akallabeth Done. The cases `BIO_C_GET_EVENT` and `BIO_C_GET_FD` are now handled together as they are unique now.
One might ask why a switch/case is not used here instead of else-ifs, however I'd rather not change it anymore because it would produce a lot of changed LOC